sky/skylet/providers/gcp/constants.py

```diff
@@ -9,6 +9,47 @@
     "mtu": 1460,
     "routingConfig": {"routingMode": "GLOBAL"},
 }
+# Required firewall rules for SkyPilot to work.
+# Note: there are more than one acceptable required rules.
+FIREWALL_RULES_REQUIRED = [
+    [
+        {
+            "direction": "INGRESS",
+            "allowed": [
+                {'IPProtocol': 'tcp', 'ports': ['0-65535']},
+                {'IPProtocol': 'udp', 'ports': ['0-65535']},
+                {'IPProtocol': 'icmp'}
+            ],
+            "sourceRanges": ["10.128.0.0/9"],
+        },
+        {
+            "direction": "INGRESS",
+            "allowed": [{
+                "IPProtocol": "tcp",
+                "ports": ["22"],
+            }],
+            "sourceRanges": ["0.0.0.0/0"],
+        }
+    ],
+    [
+        {
+            "direction": "INGRESS",
+            "allowed": [
+                {'IPProtocol': 'all'},
+            ],
+            "sourceRanges": ["10.128.0.0/9"],
+        },
+        {
+            "direction": "INGRESS",
+            "allowed": [{
+                "IPProtocol": "tcp",
+                "ports": ["22"],
```

Comment:
Is it possible this to be the following?
```
{'IPProtocol': 'all'},
```

Reply:
Looks to me it's either `all` or specifying protocol+port.
<img width="611" alt="Screen Shot 2023-01-27 at 10 45 14 AM" src="https://user-images.githubusercontent.com/1206058/215169195-28649df6-be5e-416a-bd75-915cbc0a4719.png">

Reply:
But I do think it'd be super hard to catch all cases here. let's say somehow they do
```
"IPProtocol": "tcp",
"ports": ["22", "23],
```
Then it won't match. however, there are some trade-off here. I'm not sure if it worths adding too much complexity. because we'll still create `skypilot-vpc` for skypilot to function. the only downside is users may need to move their manual firewall rules to `skypilot-vpc`. wdyt

Reply:
Do you think we can use a function like the one in https://github.com/skypilot-org/skypilot/pull/1640#discussion_r1089338295?

Reply:
yes, good point! I've updated the code to cover this case.